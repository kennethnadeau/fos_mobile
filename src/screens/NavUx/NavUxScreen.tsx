import React from 'react';
import {View, StyleSheet} from 'react-native';
import {Text} from 'react-native-elements';
import {ScreenFC} from 'react-native-navigation-register-screens';
import {SCREENS} from '@fos/constants';
import {vs, s, ms} from 'react-native-size-matters';
import {SafeAreaView} from 'react-native-safe-area-context';
import RoundedButton from '@fos/components/RoundedButton';
import Logo from '@fos/components/Logo';
import {useTranslation} from 'react-i18next';
import {Navigation} from 'react-native-navigation';

const NavUxScreen: ScreenFC = () => {
  const {t} = useTranslation();

  const headerText = t('screens.navUx.text.selectYourExperience');
  const launchAppText = t('screens.navUx.button.launchApp');
  const newSubscriptionsText = t('screens.navUx.button.newSubscriptions');
  const joinTeamInviteText = t('screens.navUx.button.joinTeamInvite');

  return (
    <SafeAreaView style={styles.container}>
      <Logo />

      <Text style={styles.header}>{headerText}</Text>

      <View style={styles.btnContainer}>
        <RoundedButton
          accessibilityLabel={launchAppText}
          accessibilityRole="button"
          accessible
          buttonStyle={styles.btn}
          onPress={() => {
            Navigation.push('STACK.ROOT', {
              component: {
                name: SCREENS.LANDING,
                options: {
                  topBar: {
                    visible: false,
                  },
                },
              },
            });
          }}
          size="large"
          title={launchAppText}
        />

        <RoundedButton
          accessibilityLabel={newSubscriptionsText}
          accessibilityRole="button"
          accessible
          buttonStyle={styles.btn}
          size="large"
          title={newSubscriptionsText}
        />

        <RoundedButton
          accessibilityLabel={joinTeamInviteText}
          accessibilityRole="button"
          accessible
          buttonStyle={styles.btn}
          size="large"
          title={joinTeamInviteText}
        />
      </View>
    </SafeAreaView>
  );
};

NavUxScreen.screenName = SCREENS.NAV_UX;

const styles = StyleSheet.create({
  btn: {
    width: ms(300),
  },
  btnContainer: {
    alignItems: 'center',
    height: vs(220),
    justifyContent: 'space-between',
    marginHorizontal: s(50),
    width: s(300),
  },
  container: {
    alignItems: 'center',
    flex: 1,
    paddingTop: vs(30),
  },
  header: {
    fontSize: s(16),
    fontWeight: '900',
    lineHeight: vs(25),
    marginVertical: vs(30),
  },
});

export default NavUxScreen;
