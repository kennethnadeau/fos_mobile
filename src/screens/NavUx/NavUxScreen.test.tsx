import React from 'react';
import NavUxScreen from './NavUxScreen';
import {render, RenderAPI} from '@fos/utils/testing';

describe('<NavUxScreen />', () => {
  let rendered: RenderAPI;

  beforeEach(() => {
    rendered = render(<NavUxScreen componentId="id" />);
  });

  it('should show logo at the top', () => {
    const actual = rendered.getByA11yLabel('App logo');

    expect(actual).toBeDefined();
  });

  it('should show header title', () => {
    const actual = rendered.queryByText('Select Your Experience');

    expect(actual).toBeDefined();
  });

  it('should show Launch App button', () => {
    const actual = rendered.getByA11yLabel('Launch App');

    expect(actual).toBeDefined();
  });

  it('should show New Subscriptions button', () => {
    const actual = rendered.getByA11yLabel('New Subscriptions');

    expect(actual).toBeDefined();
  });

  it('should show Join Team Invite button', () => {
    const actual = rendered.getByA11yLabel('Join Team Invite');

    expect(actual).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
