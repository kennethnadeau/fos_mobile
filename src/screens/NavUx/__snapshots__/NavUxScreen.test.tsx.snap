// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavUxScreen /> should match snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 58.85294117647059,
    }
  }
>
  <Image
    accessibilityLabel="App logo"
    accessible={true}
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/images/logo.png",
      }
    }
    style={
      Array [
        Object {
          "height": 213.83235294117645,
          "width": 252.85714285714286,
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      Object {
        "fontSize": 34.285714285714285,
        "fontWeight": "900",
        "lineHeight": 49.04411764705882,
        "marginVertical": 58.85294117647059,
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
  >
    Select Your Experience
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 392.35294117647055,
        "justifyContent": "space-between",
        "marginHorizontal": 107.14285714285714,
        "width": 642.8571428571429,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityLabel="Launch App"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 64.28571428571428,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 121.62941176470588,
              "justifyContent": "center",
              "padding": 8,
              "width": 471.42857142857144,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 34.285714285714285,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Launch App
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityLabel="New Subscriptions"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 64.28571428571428,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 121.62941176470588,
              "justifyContent": "center",
              "padding": 8,
              "width": 471.42857142857144,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 34.285714285714285,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            New Subscriptions
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityLabel="Join Team Invite"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 64.28571428571428,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 121.62941176470588,
              "justifyContent": "center",
              "padding": 8,
              "width": 471.42857142857144,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 34.285714285714285,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Join Team Invite
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
